django/http/multipartparser.py

```diff
@@ -166,15 +175,38 @@ def parse(self):
                 field_name = force_text(field_name, encoding, errors='replace')
 
                 if item_type == FIELD:
+                    # Avoid storing more than DATA_UPLOAD_MAX_NUMBER_FIELDS
+                    num_post_keys += 1
+                    if (settings.DATA_UPLOAD_MAX_NUMBER_FIELDS is not None and
+                            settings.DATA_UPLOAD_MAX_NUMBER_FIELDS < num_post_keys):
+                        raise TooManyFieldsSent(
+                            'The number of GET/POST parameters exceeded '
+                            'settings.DATA_UPLOAD_MAX_NUMBER_FIELDS.'
+                        )
+
+                    # Avoid reading more than DATA_UPLOAD_MAX_MEMORY_SIZE
+                    if settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None:
+                        # TODO: unclear if "+ 1" is needed... no test fail it it's removed.
```

Comment:
I don't remember anymore the specific details of this... It seems to me that the idea was to read 1 more than the max so that the max would still be valid. The test bellow is: ```data_size > settings.DATA_UPLOAD_MAX_MEMORY_SIZE```. But since we're already adding ```len(field_name) + 2``` on top of what we read this seems pointless.